diag_manager/fms_diag_axis_object.F90

```diff
@@ -556,16 +557,30 @@ subroutine get_domain_and_domain_type(axis_id, domain_type, domain, var_name)
         !! i.e a variable can have axis that are domain decomposed (x,y) and an axis that isn't (z)
         if (domain_type .eq. NO_DOMAIN .or. axis_obj(j)%type_of_domain .eq. NO_DOMAIN ) then
           !< Update the domain_type and domain, if needed
-          if (axis_obj(j)%type_of_domain .eq. TWO_D_DOMAIN .or. axis_obj(j)%type_of_domain .eq. UG_DOMAIN) then
-            domain_type = axis_obj(j)%type_of_domain
-            domain => axis_obj(j)%axis_domain
+          if ((axis_obj(j)%type_of_domain .eq. TWO_D_DOMAIN  .and. size(axis_id) > 2) &
+             & .or. axis_obj(j)%type_of_domain .eq. UG_DOMAIN) then
+               domain_type = axis_obj(j)%type_of_domain
+               domain => axis_obj(j)%axis_domain
           endif
         else
           call mpp_error(FATAL, "The variable:"//trim(var_name)//" has axis that are not in the same domain")
         endif
       endif
     enddo
   end subroutine get_domain_and_domain_type
+
+  !> @brief Gets the length of the axis based on the axis_id
+  !> @return Axis_length
+  function fms_get_axis_length(axis_id)&
```

Comment:
Why not make this an object function?  
axis_length = axis_obj(axis_id)%get_axis_length

Reply:
The model is going to call `get_axis_length`. 

`get_axis_length` currently lives in diag_axis_mod, so I added `fms_get_axis_length` as a wrapper to the object function to avoid having axis_obj code in the old diag_axis_mod.